workshops/intro_to_ML/README.md

```diff
@@ -0,0 +1,220 @@
+
+**So, how does this actually work ?**   
+We provide the machine with some sets of data (usually 2) and it is made to establish a link between the data sets or in other words, "the ML model is trained".   
+In this process of training, the machine makes an initial guess and then checks how good or bad the guess was and it calculates the loss. Then it uses an optimizer to make another optimized prediction which is better than the previous one. In other words, the loss is decreased. This process continues as many times as we specify.
+
+So, for example, we have 2 number sets as follows :   
+[ 1, 2, 3, 4, 5, 6, 7 ]   
+[ 3, 5, 7, 9, 11, 13, 15 ]   
+
+We can easily figure out the relation between the 2 given sets which is 2x + 1.   
+But how can we train a computer to do the equivalent task i.e. to figure out the relationship and predict for a future number?   
+Using data!   
+We'll be feeding it with a set of Xs, and a set of Ys, and when we train it then it should be able to figure out the relationship between them.   
+
+So, lets get started :)  
+<br/>
+Note : for this project We'll use PYTHON and a library called Tensorflow  
+<br/>
+
+# Using Google Colab
+We'll be making our project on a google colab notebook.
+Go to [colab.research.google.com](https://colab.research.google.com) and sign in with your Google account.  
+Then  Click on File > New notebook<br />
+Connect to a machine<br />
+Add a code block and we are ready to get started !
+  
+(Press <kbd>CTRL</kbd> + <kbd>Enter</kbd> to run the code)  
+<br/>
+
+# Imports
+Let's start with our imports. Imports provide us a way to exceed the limits of inbuilt modules and functions.
+They are just some pieces of code written by developers which make repetitive tasks way more easier and help in reducing our code
+To import a module (called a library) the syntax is : `import <library_name> `   
+Further, we can import something specific instead of the  complete library(to reduce program size) from the library using : `from <library_name> import <object_name>`   
+Also, we can name them something else for our ease as so that instead of accessing it from the original name, we can give it a shorter name: `import <library_name> as <easier_name>`   
+
+*For more, refer to [Imports Documentation](https://docs.python.org/3/reference/import.html)
+    
+
+Here we are importing the library ["TensorFlow"](https://www.tensorflow.org/api_docs) and calling it "tf" for ease of use.
+This library contains many tools for machine learning   
+We then import a library called ['Numpy'](https://numpy.org/doc/), which helps us to represent and manipulate our data as lists easily and quickly. We call it as "np"for ease.
+It provides a quicker and easier way to manipulate lists as compared to python's inbuilt listing system.
+Numpy documentation : 
+
+The framework for defining a neural network as a set of Sequential layers is called "keras", so we import that too.We're importing it separately for our ease
+
+```
```

Comment:
Add syntax highlighting everywhere after the 3 backticks
eg: for javascript it should be 
`(``` js)`
and similarly for HTML and CSS
`(```html)`
`(``` css)`